src/plugins/showBadgesInChat/index.tsx

```diff
@@ -0,0 +1,117 @@
+                    ))}
+                </span>
+            );
+        case "VencordContributer":
+            return isPluginDev(author.id) ? (
+                <span style={{ order: settings.store.VencordContributorPosition }}>
+                    <RoleIconComponent
+                        className={roleIconClassName}
+                        name="Vencord Contributor"
+                        size={20}
+                        src={"https://vencord.dev/assets/favicon.png"}
+                    />
+                </span>
+            ) : null;
+        case "DiscordProfile":
+            const chatBadges = discordBadges
+                .filter((badge: any) => (author.flags || author.publicFlags) & (1 << badge[0]))
+                .map((badge: any) => (
+                    <RoleIconComponent
+                        className={roleIconClassName}
+                        name={badge[1]}
+                        size={20}
+                        src={`https://cdn.discordapp.com/badge-icons/${badge[2]}.png`}
+                    />
+                ));
+            return chatBadges.length > 0 ? (
+                <span style={{ order: settings.store.DiscordProfilePosition }}>
+                    {chatBadges}
+                </span>
+            ) : null;
+        case "DiscordNitro":
+            return (author?.premiumType ?? 0) > 0 ? (
+                <span style={{ order: settings.store.DiscordNitroPosition }}>
+                    <RoleIconComponent
+                        className={roleIconClassName}
+                        name={
+                            "Discord Nitro" +
+                            (author.premiumType === 3 ? " Basic" : author.premiumType === 1 ? " Classic" : "")
+                        }
+                        size={20}
+                        src={"https://cdn.discordapp.com/badge-icons/2ba85e8026a8614b640c2837bcdfe21b.png"}
+                    />
+                </span>
+            ) : null;
+        default:
+            return null;
+    }
+}
+
+function ChatBadges({ author }: any) {
+    return (
+        <span className="vc-sbic-badge-row">
+            {settings.store.showVencordDonor && <CheckBadge badge={"VencordDonor"} author={author} />}
+            {settings.store.showVencordContributor && <CheckBadge badge={"VencordContributer"} author={author} />}
+            {settings.store.showDiscordProfile && <CheckBadge badge={"DiscordProfile"} author={author} />}
+            {settings.store.showDiscordNitro && <CheckBadge badge={"DiscordNitro"} author={author} />}
+        </span>
+    );
+}
+
+export default definePlugin({
+    name: "ShowBadgesInChat",
+    authors: [Devs.Inbestigator, Devs.KrystalSkull],
+    description: "Shows the message author's badges beside their name in chat.",
+    dependencies: ["MessageDecorationsAPI"],
+    settings,
+    start: () => {
+        addDecoration("vc-show-badges-in-chat", props => <ChatBadges author={props.message?.author} />);

```

Comment:
you use optional chaining here, which could potentially result in `author` being undefined, however the above functions expect a non null user and don't do any null safety

that makes very little sense! you should either remove this optional chaining if you're sure the author always exists
```suggestion
        addDecoration("vc-show-badges-in-chat", props => <ChatBadges author={props.message.author} />);

```
or, safer, make sure it always exists
```suggestion
        addDecoration("vc-show-badges-in-chat", props => props.message?.author ? <ChatBadges author={props.message.author} /> : null);

```